libs/blocks/gnav/gnav-contextual-search.js

```diff
@@ -0,0 +1,96 @@
+import { fetchBlogArticleIndex } from '../article-feed/article-feed.js';
+import { getArticleTaxonomy, buildArticleCard } from '../article-feed/article-helpers.js';
+import { createTag } from '../../utils/utils.js';
+
+const LIMIT = 12;
+
+function highlightTextElements(terms, elements) {
+  elements.forEach((element) => {
+    const matches = [];
+    const { textContent } = element;
+    const lowerCaseTextContent = textContent.toLowerCase();
+    terms.forEach((term) => {
+      const offset = lowerCaseTextContent.indexOf(term.toLowerCase());
+      if (offset >= 0) {
+        matches.push({ offset, term });
+      }
+    });
+    if (!matches.length) {
+      if (element.firstElementChild.nodeName === 'A') {
```

Comment:
This line causes a test failure in `test/blocks/gnav/gnav-contextual-search.test.js`: in case an element does not have any child nodes, `element.firstElementChild` returns `null`.

Reply:
I added a check, so it doesn't break.